include/aie/Dialect/AIEVec/IR/AIEVecTypeConstraints.td

```diff
@@ -101,6 +101,10 @@ class ShapesCompatibleWithContraction<string lhs, string rhs, string acc> :
 class VectorType<string name> : StrFunc<"cast<VectorType>($" # name #
                                         ".getType())">;
 
+class VectorElementType<string name> : 
```

Comment:
These already exists in upstream mlir. Check `llvm-project/mlir/include/mlir/IR/OpBase.td`. There's a lot of stuff there you'll find helpful.